src/lib/features/listening-history/model/store.ts

```diff
@@ -12,14 +26,31 @@ export const listeningHistory = persistentWritable<number[]>([], {
  * Function that adds a podcast id to the beginning of listening history.
  * It removes any previous occurrences of the same podcast.
  */
-export const addToListeningHistory = (podcastId: number) => {
+const addToLocalListeningHistory = (podcastId: number) => {
   listeningHistory.update((history) => {
     return [podcastId, ...history.filter((id) => id !== podcastId)].slice(0, 6);
   });
 };
 
+/**
+ * Function that adds a podcast id into `history` SB table.
+ */
+const addToCloudListeningHistory = async (podcastId: number) => {
+  if (!get(user)) return;
+  await supabaseClient.from('history').insert([
```

Comment:
issue: this is an unhandled promise, it might throw, for example, when we are offline

suggestion: remove the check for the user above and just wrap the `await` in a `try-catch` to prevent both unauthenticated requests and any network errors

Reply:
this also applies to several other places in the code